File: yht0827/1월/1월 28일/방금 그곡.java
class Song {
	int time;
	String title;
	String lylics;
}

class solution {
	public String solution(String m, String[] musicinfos) {
		List<Song> list = new ArrayList<Song>();

		m = replaceString(m);

		for (int i = 0; i < musicinfos.length; i++) {
			String[] musicInfo = musicinfos[i].split(",");

			String[] start = musicInfo[0].split(":");
			String[] end = musicInfo[1].split(":");
			Song song = new Song();
			String title = musicInfo[2];
			String contents = replaceString(musicInfo[3]);

			song.title = title;

			int sH = Integer.parseInt(start[0]);
			int eH = Integer.parseInt(end[0]);
			int diff = (eH - sH) * 60;

			int sM = Integer.parseInt(start[1]);
			int eM = Integer.parseInt(end[1]);

			int minute = (eM + diff) - sM;
			song.time = minute;

			String total = "";

			int div = minute / contents.length();

			for (int j = 0; j < div; j++) {
				total += contents;
			}

			if (!"".equals(total) && minute % contents.length() != 0)
				total += contents.substring(0, minute % contents.length());

			if ("".equals(total))
				total = contents.substring(0, minute);

			song.lylics = total;

			list.add(song);
		}

		int maxTime = Integer.MIN_VALUE;
		String maxTitle = "";

		for (int i = 0; i < list.size(); i++) {
			Song song = list.get(i);

			if (song.lylics.contains(m)) {
				if (maxTime < song.time) {
					maxTitle = song.title;
					maxTime = Math.max(maxTime, song.time);
				}
			}

		}

		if (!"".equals(maxTitle))
			return maxTitle;

		return "(None)";
	}

	public String replaceString(String s) {
		return s.replaceAll("A#", "V")
			.replaceAll("C#", "W")
			.replaceAll("D#", "X")
			.replaceAll("F#", "Y")
			.replaceAll("G#", "Z");
	}
}

File: Eunleelee/2024-01/1월 28일/방금그곡.py
import math

def solution(m, musicinfos):
    m = m.replace('A#', 'H').replace('C#', 'I').replace('D#', 'J').replace('F#', 'K').replace('G#', 'L')
    music_info = []
    
    for num, musicinfo in enumerate(musicinfos):
        start, end, name, music = musicinfo.split(',')   
        
        music = music.replace('A#', 'H').replace('C#', 'I').replace('D#', 'J').replace('F#', 'K').replace('G#', 'L')
        music_len = (int(end[:2]) * 60 + int(end[3:])) - (int(start[:2]) * 60 + int(start[3:]))
        
        if len(music) < music_len:
            music += music * (music_len // len(music)) + music[:(music_len % len(music))]
        elif len(music) > music_len:
            music = music[:music_len]
        
        if m in music:
            music_info.append([music_len, name, num])
            
    if len(music_info) == 0:
        return "(None)"
    
    if len(music_info) != 1:
        music_info = sorted(music_info, key = lambda x : (-x[0], x[2]))
  
    return music_info[0][1]